client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,476 @@
+                    "Waiting for next block, {transactions_to_confirm} transactions pending..."
+                ),
+            );
+        }
+
+        // wait till all transactions are confirmed or we have surpassed max processing age for the last sent transaction
+        while !unconfirmed_transasction_map.is_empty()
+            && current_blockheight.load(Ordering::Relaxed) < max_valid_block_height
+        {
+            let blockheight = current_blockheight.load(Ordering::Relaxed);
+
+            if let Some(progress_bar) = progress_bar {
+                set_message_for_confirmed_transactions(
+                    progress_bar,
+                    num_confirmed_transactions.load(std::sync::atomic::Ordering::Relaxed),
+                    total_transactions,
+                    Some(blockheight),
+                    max_valid_block_height,
+                    "Checking transaction status...",
+                );
+            }
+
+            if let Some(tpu_client) = tpu_client {
+                let instant = Instant::now();
+                // retry sending transaction only over TPU port / any transactions sent over RPC will be automatically rebroadcast by the RPC server
+                let txs_to_resend_over_tpu = unconfirmed_transasction_map
+                    .iter()
+                    .filter(|x| blockheight < x.last_valid_blockheight)
+                    .map(|x| x.serialized_transaction.clone())
+                    .collect();
+                let _ = tpu_client
+                    .try_send_wire_transaction_batch(txs_to_resend_over_tpu)
+                    .await;
+
+                let elapsed = instant.elapsed();
+                if elapsed < TPU_RESEND_REFRESH_RATE {
+                    tokio::time::sleep(TPU_RESEND_REFRESH_RATE - elapsed).await;
+                }
+            } else {
+                tokio::time::sleep(Duration::from_millis(100)).await;
+            }
+            if let Some(max_valid_block_height_in_remaining_transaction) =
+                unconfirmed_transasction_map
+                    .iter()
+                    .map(|x| x.last_valid_blockheight)
+                    .max()
+            {
+                max_valid_block_height = max_valid_block_height_in_remaining_transaction;
+            }
+        }
+
+        if let Some(progress_bar) = progress_bar {
+            set_message_for_confirmed_transactions(
+                progress_bar,
+                num_confirmed_transactions.load(std::sync::atomic::Ordering::Relaxed),
+                total_transactions,
+                Some(current_blockheight.load(Ordering::Relaxed)),
+                max_valid_block_height,
+                "Checking transaction status...",
+            );
+        }
+    }
+}
+
+// This is a new method which will be able to send and confirm a large amount of transactions
+// usually the send_and_confirm_transactions_with_spinner for QUIC clients are optimized for small number of transactions
```

Comment:
```suggestion
/// It is preferable to be used instead of `send_and_confirm_transactions_with_spinner` if the number of transactions is large.
/// Typical use-case is deployment of a large program.
```